ci: quickbuild failure

Quickbuild failure below, I would guess this stems from the recent upgrade of Ubuntu on the quickbuild environment. cc @jszakmeister 

@bfredl IIRC you put a workaround in place for this, but maybe it doesn't apply to Release mode?

```
7:22:41,607 WARN  - In file included from /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/bufhl_defs.h:6:0,
07:22:41,607 WARN  - from /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/map.h:9,
07:22:41,607 WARN  - from /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/buffer_defs.h:40,
07:22:41,607 WARN  - from /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/normal.h:6,
07:22:41,607 WARN  - from /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/ex_cmds_defs.h:8,
07:22:41,607 WARN  - from /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/ex_eval.h:5,
07:22:41,607 WARN  - from /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/globals.h:8,
07:22:41,607 WARN  - from /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/vim.h:291,
07:22:41,607 WARN  - from /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/api/private/handle.h:4,
07:22:41,607 WARN  - from /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/buffer.c:26:
07:22:41,607 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/buffer.c: In function ‘bufhl_mark_adjust’:
07:22:41,608 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/lib/kbtree.h:238:28: error: array subscript is below array bounds [-Werror=array-bounds]
07:22:41,608 WARN  - kp = __KB_KEY(key_t, x)[i]; \
07:22:41,608 WARN  - ^
07:22:41,608 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/lib/kbtree.h:418:2: note: in expansion of macro ‘__KB_DEL’
07:22:41,608 WARN  - __KB_DEL(name, key_t, kbnode_t, T) \
07:22:41,608 WARN  - ^
07:22:41,608 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/lib/kbtree.h:391:3: note: in expansion of macro ‘KBTREE_INIT_IMPL’
07:22:41,608 WARN  - KBTREE_INIT_IMPL(name, key_t, kbnode_##name##_t, __cmp, T, (sizeof(kbnode_##name##_t)+(2*T)*sizeof(void *)))
07:22:41,608 WARN  - ^
07:22:41,608 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/bufhl_defs.h:32:1: note: in expansion of macro ‘KBTREE_INIT’
07:22:41,608 WARN  - KBTREE_INIT(bufhl, BufhlLine *, BUFHL_CMP, 10) // -V512
07:22:41,608 WARN  - ^
07:22:41,608 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/lib/kbtree.h:239:23: error: array subscript is below array bounds [-Werror=array-bounds]
07:22:41,608 WARN  - __KB_KEY(key_t, x)[i] = __kb_delp_aux_##name(b, xp, 0, 1); \
07:22:41,608 WARN  - ^
07:22:41,608 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/lib/kbtree.h:418:2: note: in expansion of macro ‘__KB_DEL’
07:22:41,608 WARN  - __KB_DEL(name, key_t, kbnode_t, T) \
07:22:41,608 WARN  - ^
07:22:41,608 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/lib/kbtree.h:391:3: note: in expansion of macro ‘KBTREE_INIT_IMPL’
07:22:41,608 WARN  - KBTREE_INIT_IMPL(name, key_t, kbnode_##name##_t, __cmp, T, (sizeof(kbnode_##name##_t)+(2*T)*sizeof(void *)))
07:22:41,608 WARN  - ^
07:22:41,608 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/bufhl_defs.h:32:1: note: in expansion of macro ‘KBTREE_INIT’
07:22:41,608 WARN  - KBTREE_INIT(bufhl, BufhlLine *, BUFHL_CMP, 10) // -V512
07:22:41,608 WARN  - ^
07:22:41,608 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/lib/kbtree.h:243:28: error: array subscript is below array bounds [-Werror=array-bounds]
07:22:41,608 WARN  - kp = __KB_KEY(key_t, x)[i]; \
07:22:41,608 WARN  - ^
07:22:41,608 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/lib/kbtree.h:418:2: note: in expansion of macro ‘__KB_DEL’
07:22:41,608 WARN  - __KB_DEL(name, key_t, kbnode_t, T) \
07:22:41,608 WARN  - ^
07:22:41,608 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/lib/kbtree.h:391:3: note: in expansion of macro ‘KBTREE_INIT_IMPL’
07:22:41,608 WARN  - KBTREE_INIT_IMPL(name, key_t, kbnode_##name##_t, __cmp, T, (sizeof(kbnode_##name##_t)+(2*T)*sizeof(void *)))
07:22:41,608 WARN  - ^
07:22:41,608 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/bufhl_defs.h:32:1: note: in expansion of macro ‘KBTREE_INIT’
07:22:41,608 WARN  - KBTREE_INIT(bufhl, BufhlLine *, BUFHL_CMP, 10) // -V512
07:22:41,608 WARN  - ^
07:22:41,608 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/lib/kbtree.h:244:23: error: array subscript is below array bounds [-Werror=array-bounds]
07:22:41,608 WARN  - __KB_KEY(key_t, x)[i] = __kb_delp_aux_##name(b, xp, 0, 2); \
07:22:41,608 WARN  - ^
07:22:41,608 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/lib/kbtree.h:418:2: note: in expansion of macro ‘__KB_DEL’
07:22:41,608 WARN  - __KB_DEL(name, key_t, kbnode_t, T) \
07:22:41,608 WARN  - ^
07:22:41,608 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/lib/kbtree.h:391:3: note: in expansion of macro ‘KBTREE_INIT_IMPL’
07:22:41,608 WARN  - KBTREE_INIT_IMPL(name, key_t, kbnode_##name##_t, __cmp, T, (sizeof(kbnode_##name##_t)+(2*T)*sizeof(void *)))
07:22:41,608 WARN  - ^
07:22:41,608 WARN  - /home/quickbuild/buildagent/workspace/root/neovim/pull-requests-automated/src/nvim/bufhl_defs.h:32:1: note: in expansion of macro ‘KBTREE_INIT’
07:22:41,608 WARN  - KBTREE_INIT(bufhl, BufhlLine *, BUFHL_CMP, 10) // -V512
07:22:41,608 WARN  - ^
07:22:41,900 WARN  - cc1: all warnings being treated as errors
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -250,13 +250,14 @@ if(HAS_DIAG_COLOR_FLAG)
 endif()
 
 if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-  if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.8.5")
-    # Array-bounds testing is broken in some GCC versions before 4.8.5.
-    # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=56273
-    check_c_compiler_flag(-Wno-array-bounds HAS_NO_ARRAY_BOUNDS_FLAG)
-    if(HAS_NO_ARRAY_BOUNDS_FLAG)
-      add_definitions(-Wno-array-bounds)
-    endif()
+  # 1. Array-bounds testing is broken in some GCC versions before 4.8.5.
+  #    https://gcc.gnu.org/bugzilla/show_bug.cgi?id=56273
+  # 2. But _Pragma("...ignored") is broken (unresolved) in GCC 5+:
+  #    https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66099
+  # So we must disable -Warray-bounds globally for GCC (for kbtree.h, #7083).
+  check_c_compiler_flag(-Wno-array-bounds HAS_NO_ARRAY_BOUNDS_FLAG)
+  if(HAS_NO_ARRAY_BOUNDS_FLAG)
+    add_definitions(-Wno-array-bounds)
   endif()
 endif()
 

--- a/src/nvim/lib/kbtree.h
+++ b/src/nvim/lib/kbtree.h
@@ -390,34 +390,14 @@
 #define KBTREE_INIT(name, key_t, __cmp, T) \
   KBTREE_INIT_IMPL(name, key_t, kbnode_##name##_t, __cmp, T, (sizeof(kbnode_##name##_t)+(2*T)*sizeof(void *)))
 
-#if (!defined(__clang__) && !defined(__INTEL_COMPILER)) &&  (__GNUC__ > 4 )
-
-// The index trickery shouldn't be UB anymore,
-// still it is to much for gcc:s -Werror=array-bounds
-# define __KB_PRAGMA_START \
-    _Pragma("GCC diagnostic push") \
-    _Pragma("GCC diagnostic ignored \"-Warray-bounds\"")
-
-# define __KB_PRAGMA_END \
-    _Pragma("GCC diagnostic pop") \
-
-#else
-
-# define __KB_PRAGMA_START
-# define __KB_PRAGMA_END
-
-#endif
-
 #define KBTREE_INIT_IMPL(name, key_t, kbnode_t, __cmp, T, ILEN)			\
-	__KB_PRAGMA_START \
 	__KB_TREE_T(name, key_t, T)							\
 	__KB_GET_AUX1(name, key_t, kbnode_t, __cmp)			\
 	__KB_GET(name, key_t, kbnode_t)						\
 	__KB_INTERVAL(name, key_t, kbnode_t)					\
 	__KB_PUT(name, key_t, kbnode_t, __cmp, T, ILEN)				\
 	__KB_DEL(name, key_t, kbnode_t, T) \
-	__KB_ITR(name, key_t, kbnode_t) \
-	__KB_PRAGMA_END
+	__KB_ITR(name, key_t, kbnode_t)
 
 #define KB_DEFAULT_SIZE 512
 

